src/main/java/database_backend/DatabaseAccessor.java

```diff
@@ -0,0 +1,28 @@
+package database_backend;
+
+import entities.DrugInfo;
+import entities.DrugList;
+import search_database_use_case.IDrugListAccessor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Handles all database de/serialization.
+ */
+public class DatabaseAccessor implements IDrugListAccessor {
+    /**
+     * If necessary, deserializes drug info and updates {@link DrugList} with the resulting list of {@link DrugInfo}
+     * objects.
+     * @return The deserialized {@link List<DrugInfo>}.
+     */
+    public List<DrugInfo> getDrugList() {
+        List<DrugInfo> list = DrugList.getDrugList();
+
+        if (list == null || list.isEmpty()) {
+            // todo deserialize drug info here
+        }
```

Comment:
I guess serialization is fine but I have to see if it will be compatible with how I stored the data in csv.

Reply:
Yeah, by deserialization I meant reading the data from the csv and formatting it such that I can implement the search functions I want.